### PR TITLE
Online DDL: support 'SHOW VITESS_MIGRATIONS' queries

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Supporting these queries in `vtgate`:

- `SHOW VITESS_MIGRATIONS` - shows all migrations for keyspace
- `SHOW VITESS_MIGRATIONS LIKE '...'` - shows vitess migrations filtered for UUID, or context, or status. For example:
  - `SHOW VITESS_MIGRATIONS LIKE '6df3201f_7fda_11eb_aa27_f875a4d24e90'` - shows a specific migration
  - `SHOW VITESS_MIGRATIONS LIKE 'failed'` - show migrations with `failed` status 
- `SHOW VITESS_MIGRATIONS WHERE ...` - add any custom filter, e.g.:
  - `SHOW VITESS_MIGRATIONS WHERE started_timestamp > NOW() - INTERVAL 7 DAY`
- `SHOW VITESS_MIGRATIONS FROM keyspace [LIKE ...|WHERE ...]` - explicitly indicate keyspace

## Related Issue(s)

- https://github.com/vitessio/vitess/issues/7604
- #6926 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

## Impacted Areas in Vitess
Components that this PR will affect:

- [x]  Query Serving
- [ ]  VReplication
- [ ]  Cluster Management
- [ ]  Build/CI
- [ ]  VTAdmin

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Seems like even though we specify `key.DestinationAllShards`, query only runs on a single shard.

**Comment by shlomi-noach** [comment]
Thank you very much to @harshit-gangal and @systay who guided and led me in the right path, and for noticing my Online DDL `endtoend` tests had wrong/broken shard configuration!

**Comment by shlomi-noach** [comment]
This is now ready to review. This PR only introduced `SHOW VITESS_MIGRATIONS` command(s), which is a more urgent requirement. `ALTER VITESS_MIGRATIONS` will be handled in a followup PR.

Documentation to follow.

**Comment by shlomi-noach** [comment]
Added one last commit, refactored a couple functions in `endtoend` tests, where I originally did a lot of copy+paste. There's more cleanup work to be done, but not in this PR.

**Review by systay — APPROVED** [review]

File: go/test/endtoend/cluster/query_util.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import (
+	"io"
+	"strings"
+
+	"vitess.io/vitess/go/sqltypes"
+
+	"github.com/olekukonko/tablewriter"
+)
+
+// printQueryResult will pretty-print a QueryResult to the logger.
```
Comment (systay) [inline go/test/endtoend/cluster/query_util.go:28]: nit: the comment should have the same case for the method name

Comment (shlomi-noach) [reply go/test/endtoend/cluster/query_util.go]: fixed

File: go/test/endtoend/cluster/query_util.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Vitess Authors.
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/test/endtoend/cluster/query_util.go]: fixed

File: go/vt/sqlparser/ast.go [hunk 3]
```diff
@@ -2480,7 +2480,7 @@ func (node *Show) Format(buf *TrackedBuffer) {
 func (node *ShowLegacy) Format(buf *TrackedBuffer) {
 	nodeType := strings.ToLower(node.Type)
 	if (nodeType == "tables" || nodeType == "columns" || nodeType == "fields" || nodeType == "index" || nodeType == "keys" || nodeType == "indexes" ||
-		nodeType == "databases" || nodeType == "schemas" || nodeType == "keyspaces" || nodeType == "vitess_keyspaces" || nodeType == "vitess_shards" || nodeType == "vitess_tablets") && node.ShowTablesOpt != nil {
+		nodeType == "databases" || nodeType == "schemas" || nodeType == "keyspaces" || nodeType == "vitess_keyspaces" || nodeType == "vitess_shards" || nodeType == "vitess_tablets" || nodeType == "vitess_migrations") && node.ShowTablesOpt != nil {
```
Comment (harshit-gangal) [inline go/vt/sqlparser/ast.go:2483]: this is not needed as show vitess migrations is not legacy.

**Review by harshit-gangal — COMMENTED** [review]
I would suggest to refactor the file directory as `go/test/endtoend/onlineddl/<ghost/vrepl/...>` and place query_util and vtgate_util under  `go/test/endtoend/onlineddl/` those are the only tests that are using it.

File: go/vt/sqlparser/ast.go (hunk 3 above)
Comment (shlomi-noach) [reply go/vt/sqlparser/ast.go]: removed

**Comment by shlomi-noach** [comment]
@harshit-gangal I have restructured the tests, all nested under `go/test/endtoend/onlineddl`
```shell
go/test/endtoend/onlineddl # base package with shared functionality
go/test/endtoend/onlineddl/vrepl
go/test/endtoend/onlineddl/vrepl_stress
go/test/endtoend/onlineddl/ghost
go/test/endtoend/onlineddl/revert
```

**Pull request merged**